CS update after upstream changes

jrfnl opened the pull request:
### PHPCS ruleset: update ruleset for upstream changes

* Fix the name and description to prevent confusion between the project ruleset and the organisation ruleset.
* Set the minimum PHP version for the PHPCompatibility standard.
* Ignore two PHPCompatibility non-issues.
* Don't require property type declarations.
* Ensure special characters used as literals in an exclude pattern are escaped.

### CS: no whitespace before return type colon

### CS: blank line between different use statement types

### CS: import all things

### CS: miscellaneous other whitespace fixes

jrfnl committed:
PHPCS ruleset: update ruleset for upstream changes

* Fix the name and description to prevent confusion between the project ruleset and the organisation ruleset.
* Set the minimum PHP version for the PHPCompatibility standard.
* Ignore two PHPCompatibility non-issues.
* Don't require property type declarations.
* Ensure special characters used as literals in an exclude pattern are escaped.

jrfnl committed:
CS: no whitespace before return type colon

jrfnl committed:
CS: blank line between different use statement types

jrfnl committed:
CS: import all things

jrfnl committed:
CS: miscellaneous other whitespace fixes

jaapio commented:
Looks like there is one autofix round missing for `SlevomatCodingStandard.Commenting.DocCommentSpacing.IncorrectAnnotationsGroup` :-)

jrfnl commented:
@jaapio Yup... little weird as the build passed locally, but then again, some of the local installs I have of the coding standards may be out of date. Will have a look.

jrfnl committed:
CS: docblock tag grouping

Apparently Doctrine wants the related `@psalm` annotations to be grouped with the `@param` or `@return` tags.

jrfnl commented:
Looks like there was a hick-up in the GHA run. @jaapio Could you restart the build ?
